src/Microsoft.ML.Data/Evaluators/MamlEvaluator.cs

```diff
@@ -239,23 +239,23 @@ private IDataView WrapPerInstance(RoleMappedData perInst)
                 colsToKeep.Add(MetricKinds.ColumnNames.FoldIndex);
 
             // Maml always outputs a name column, if it doesn't exist add a GenerateNumberTransform.
-            if (perInst.Schema.Name == null)
+            if (perInst.Schema.Name?.Name is string nameName)
```

Comment:
This is sort of a fun application of pattern matching. If that evaluated to `null`, then the pattern would not match. Consider this small program:

```csharp
private static void TestString(string s)
{
    if (s is string ss)
        Console.WriteLine($"Yup {ss}");
    else
        Console.WriteLine("Nope");
}

private static void Main(string[] args)
{
    TestString(null);
    TestString("hi");
}
```

The output is:

```
Nope
Yup hi
```

Very good. Yet, let's imagine I change to this:

```csharp
    if (s is var ss)
```

Now, that `ss` is still of `string` type, but now the outcome becomes this:

```
Yup
Yup hi
```

I don't fully appreciate in what way this `var` pattern matching matches the type, yet evinces such very different behavior! No doubt one of the donut people knows more about this.